src/bio/bio_wal.c

```diff
@@ -2014,15 +2028,35 @@ meta_close(struct bio_meta_context *mc)
 int
 meta_open(struct bio_meta_context *mc)
 {
-	int	rc;
+	struct meta_header *hdr = &mc->mc_meta_hdr;
+	int                 rc;
 
 	rc = meta_csum_init(mc, HASH_TYPE_CRC32);
 	if (rc)
 		return rc;
 
 	rc = load_meta_header(mc);
-	if (rc)
+	if (rc) {
 		meta_csum_fini(mc);
+		return rc;
+	}
+
+	/* Auto upgrade meta header from version 1 to 2 */
+	if (hdr->mh_version == 1) {
+		struct meta_header_v1 *v1 = (struct meta_header_v1 *)hdr;
+
+		v1->mh_csum     = 0;
+		hdr->mh_version = BIO_META_VERSION;
+		uuid_copy(hdr->mh_pool_id, mc->mc_meta->bic_pool_id);
+
+		rc = write_header(mc, mc->mc_meta, hdr, sizeof(*hdr), &hdr->mh_csum);
```

Comment:
Is there anything placed directly after the metadata header in the blob in v1?

Reply:
No, the whole 4k block is used for header.